src/raster/r.niche.similarity/r.niche.similarity.py

```diff
@@ -88,8 +87,8 @@ def tmpname(prefix):
     """Generate a tmp name which contains prefix, store the name in the
     global list.
     """
-    tmpf = prefix + str(uuid.uuid4())
-    tmpf = string.replace(tmpf, "-", "_")
+    tmpf = 'prefix{}'.format(uuid.uuid4())
+    tmpf = tmpf.replace("-", "_")
```

Comment:
Perhaps this function could be used instead:

https://grass.osgeo.org/grass82/manuals/libpython/script.html#script.utils.append_uuid